compiler/rustc_middle/src/util/call_kind.rs

```diff
@@ -129,6 +132,8 @@ pub fn call_kind<'tcx>(
             && fn_call_span.desugaring_kind() == Some(DesugaringKind::TryBlock)
         {
             Some((CallDesugaringKind::TryBlockFromOutput, method_substs.type_at(0)))
+        } else if fn_call_span.is_desugaring(DesugaringKind::Await) {
+            Some((CallDesugaringKind::Await, method_substs.type_at(0)))
```

Comment:
Seems somewhat unexpected to have this under `CallKind::Normal`. The docs say `CallKind::Normal` corresponds to method calls. I mean this is consistent with previous usage, but maybe we should introduce something like `CallKind::Other`?

Reply:
Hmm... this just feels like uplifting the `desugaring` part of the `Normal` variant up to its own variant. I don't see how this makes things much cleaner :sweat_smile:

Reply:
I just meant that this would be more consistent with the docs of `CallKind::Normal`. Feel free to ignore since `CallKind::Normal` was also used for other `CallDesugaringKind`s before for which that variant doesn't really match the docs afaict. The rest of the PR looks good to me, so r=me.